soc/espressif/esp32s3/esp32s3-mp.c

```diff
@@ -33,9 +34,27 @@ void esp_appcpu_start(void *entry_point)
 		REG_CLR_BIT(SYSTEM_CORE_1_CONTROL_0_REG, SYSTEM_CONTROL_CORE_1_RESETTING);
 	}
 
-	esp_rom_ets_set_appcpu_boot_addr((void *)entry_point);
-
+	/* THIS IS REQUIRED FOR AMP RELIABLE
+	 * OPERATION AS WELL, PLEASE DON'T touch on the dummy write below!
+	 *
+	 * Note that the logging done here is ACTUALLY REQUIRED FOR RELIABLE
+	 * OPERATION!  At least one particular board will experience spurious
+	 * hangs during initialization (usually the APPCPU fails to start at
+	 * all) without these calls present.  It's not just time -- careful
+	 * use of k_busy_wait() (and even hand-crafted timer loops using the
+	 * Xtensa timer SRs directly) that duplicates the timing exactly still
+	 * sees hangs.  Something is happening inside the ROM UART code that
+	 * magically makes the startup sequence reliable.
+	 *
+	 * Leave this in place until the sequence is understood better.
```

Comment:
and two years later you have still not understood the startup sequence better and know why this is needed ?
Ref: https://github.com/zephyrproject-rtos/zephyr/pull/51314

Reply:
Uhm yes? There is a point that the uart0 rom code does an extra work that releases the appcpu.

Unfortunately I don't work for espressif and I don't have access to the rom code or the developer team.

Reply:
Yes, I don't have access to that documentation and the extra work that is done by the uart0.

Reply:
Can you provide a link to the espidf code that does this?